tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -201,3 +211,54 @@ void ROOT::Experimental::Detail::RPageSink::CommitCluster(ROOT::Experimental::NT
    ++fLastClusterId;
    fPrevClusterNEntries = nEntries;
 }
+
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::Detail::RPageSink::Merge(
+   const std::vector<std::unique_ptr<RPageSource>>& sources)
+{
+   std::cout << "\n\ngot " << sources.size() << " files to merge\n";
+   if (sources.empty()) {
+      return RResult<void>::Success();
+   }
+
+   // todo(max) handle non-empty PageSink NTuple descriptor
+   sources.front()->Attach();
+   const auto& sourceDesc = sources.front()->GetDescriptor();
+   sourceDesc.PrintInfo(std::cout);
+
+   std::cout << "input has " << sourceDesc.GetColumnIds().size() << " columns\n";
+
+   // assume all inputs have the same size header and footer
+   auto szFooter = sourceDesc.GetFooterSize();
+   auto szHeader = sourceDesc.GetHeaderSize();
+   std::cout << "source header size " << szHeader << "\n";
+   std::cout << "source footer size " << szFooter << "\n";
+
+   // fill header to get column ids
+   auto headerBuffer = std::make_unique<unsigned char[]>(szHeader);
+   sourceDesc.SerializeHeader(headerBuffer.get());
+   fDescriptorBuilder.SetFromHeader(headerBuffer.get());
```

Comment:
Here I make a copy of the first RNTuple source header and use it as the merged RNTuple header. This is a bit hacky, but I think it's OK, as long as we override the RNTuple `uuid` at a later point (not implemented yet).  It is a workaround for not having `RColumn` objects to use with `RPageSink::AddColumn(DescriptorId_t fieldId, const RColumn &column)`